core/trino-main/src/main/java/io/trino/sql/planner/plan/TableFunctionProcessorNode.java

```diff
@@ -50,6 +50,8 @@ public class TableFunctionProcessorNode
     // pruneWhenEmpty is false if and only if all original input tables are KEEP WHEN EMPTY
     private final boolean pruneWhenEmpty;
 
+    private final boolean preferStreaming;
```

Comment:
This property should be derived in the operator. At least until we have separate "physical" PlanNodes to represent the streaming or buffering TableFunction.

In case you wonder how we can know in the LocalExecutionPlanner if the node is row semantics or set semantics: we can tell by the `specification`. For row semantics it is empty.

Reply:
I initially assumed it - but I was confused if it is the right approach

Reply:
But how would we identify between row semantics and set semantics which doesn't have any partitioning specified ? We wanted to focus only on row semantics as of this PR.